cmake/alpakaCommon.cmake

```diff
@@ -261,278 +261,146 @@ endif()
 #-------------------------------------------------------------------------------
 # Find CUDA.
 if(ALPAKA_ACC_GPU_CUDA_ENABLE)
+    include(CheckLanguage)
+    check_language(CUDA)
 
-    if(NOT DEFINED ALPAKA_CUDA_VERSION)
-        set(ALPAKA_CUDA_VERSION 9.0)
-    endif()
-
-    if(ALPAKA_CUDA_VERSION VERSION_LESS 9.0)
-        message(FATAL_ERROR "CUDA Toolkit < 9.0 is not supported!")
-
-    else()
-        find_package(CUDA "${ALPAKA_CUDA_VERSION}")
-        if(NOT CUDA_FOUND)
-            message(FATAL_ERROR "Optional alpaka dependency CUDA could not be found!")
-        else()
-            set(ALPAKA_CUDA_VERSION "${CUDA_VERSION}")
-            if(CUDA_VERSION VERSION_LESS 10.3)
-                set(ALPAKA_CUDA_ARCH "30" CACHE STRING "GPU architecture")
-            else()
-                set(ALPAKA_CUDA_ARCH "35" CACHE STRING "GPU architecture")
-            endif()
-            set(ALPAKA_CUDA_COMPILER "nvcc" CACHE STRING "CUDA compiler")
-            set_property(CACHE ALPAKA_CUDA_COMPILER PROPERTY STRINGS "nvcc;clang")
+    if(CMAKE_CUDA_COMPILER)
+        enable_language(CUDA)
+        find_package(CUDAToolkit ${ALPAKA_CUDA_VERSION} REQUIRED)
```

Comment:
Which CUDAToolkit version is searched here, if I do not set `ALPAKA_CUDA_VERSION`? Will this even error?

Reply:
See above